packages/engine/src/scene/functions/loadGLTFModel.ts

```diff
@@ -201,11 +201,21 @@ export const proxifyParentChildRelationships = (obj: Object3D) => {
     children: {
       get() {
         if (EngineRenderer.instance?.rendering) return []
-        return hasComponent(objEntity, EntityTreeComponent)
-          ? getComponent(objEntity, EntityTreeComponent)
-              .children.filter((child) => getOptionalComponent(child, GroupComponent)?.length)
-              .flatMap((child) => getComponent(child, GroupComponent))
-          : []
+        if (hasComponent(objEntity, EntityTreeComponent)) {
+          const childEntities = getComponent(objEntity, EntityTreeComponent).children.filter((child) =>
+            hasComponent(child, SceneObjectComponent)
+          )
+          const result: Object3D[] = []
+          for (const childEntity of childEntities) {
+            if (!hasComponent(childEntity, GroupComponent)) continue
+            const objs = getComponent(childEntity, GroupComponent)
+            if (objs.length === 0) continue
+            result.push(objs[0])
```

Comment:
this is going to lead to problems as some entities have more than one object in the group

Reply:
Yes but now that everything is one entity per Object3D, this shouldn't lead to any missed elements in the hierarchy, except for helpers and ancillary objects (such as lights)

Reply:
Is there a situation somewhere in the engine where we're using the proxified "children" property to grab multiple object3Ds from the same entity?

Reply:
Again, are there any situations where we're using the proxified "children" property to grab these helpers or extra objects, instead of directly querying the group component through the ECS?
At this point, it seems the only time we're really utilizing the proxified "children" property is when we're doing gltf exports, or grabbing the first element of the group component (like for animations) just as is happening in this code